server/apps/archive/archive.py

```diff
@@ -412,6 +413,78 @@ def can_edit(self, item, user_id):
 
         return True, ''
 
+    def remove_expired(self, doc):
+        """
+        Removes the article from production if the state is spiked
+        """
+
+        assert doc[config.CONTENT_STATE] == 'spiked', \
+            "Article state is %s. Only Spiked Articles can be removed" % doc[config.CONTENT_STATE]
+
+        doc_id = str(doc[config.ID_FIELD])
+        super().delete_action({config.ID_FIELD: doc_id})
+        get_resource_service('archive_versions').delete(lookup={versioned_id_field(): doc_id})
+
+        packages = self.get_packages(doc_id)
+        if packages.count() > 0:
+            processed_packages = []
+            for package in packages:
+                if str(package[config.ID_FIELD]) not in processed_packages:
+                    processed_packages.extend(self.remove_refs_in_package(package, doc_id, processed_packages))
+
+    def get_packages(self, doc_id):
+        """
+        Retrieves if an article identified by doc_id is referenced in a package.
+        :return: articles of type composite
+        """
+
+        query = {'query': {'filtered': {'filter': {'and': [{'term': {'type': 'composite'}}]},
```

Comment:
I think we should fetch the data from Mongo in all the background jobs.
The spiked document already has a reference to all the packages in which is it included into, so we should be able to fetch the packages from Mongo by `_id`.